### PR TITLE
re-do cargo-miri host/target detection logic to match rustbuild

### PR DESCRIPTION
**Pull request by RalfJung**
@oli-obk I think that's better than looking at `--emit` like we did before... what do you think?

### CODE REVIEW

File: src/bin/cargo-miri.rs [hunk 1]
```diff
@@ -491,24 +499,21 @@ fn in_cargo_miri() {
...
+    /// Determines if we are being invoked (as rustc) to build a crate for
+    /// the "target" architecture, in contrast to the "host" architecture.
+    /// Host crates are for build scripts and proc macros and still need to
+    /// be built like normal; target crates need to be built for or interpreted
+    /// by Miri.
     ///
-    /// Currently, we detect this by checking for "--emit=link",
-    /// which indicates that Cargo instruced rustc to output
-    /// a native object.
+    /// Currently, we detect this by checking for "--target=", which flag is
```
Comment (oli-obk) [inline src/bin/cargo-miri.rs:508]:
```suggestion
    /// Currently, we detect this by checking for "--target=", which is
```

**Comment by oli-obk** [comment]
Yea, this looks more robust.

**Comment by RalfJung** [comment]
While at it I also made it honor the RUSTC env var.

**Pull request merged**